com.unity.netcode.gameobjects/Tests/Runtime/RpcQueueTests.cs

```diff
@@ -110,7 +110,9 @@ public void RpcQueueContainerClass()
             var rpcQueueContainer = new MessageQueueContainer(NetworkManagerHelper.NetworkManagerObject, 0, true);
 
             // Make sure we set testing mode so we don't try to invoke RPCs
+#if UNITY_EDITOR || DEVELOPMENT_BUILD
```

Comment:
When I see "testing state" I think INCLUDE_UNITY_TESTS, not DEVELOPMENT_BUILD.... is this used outside of "unit test" passes?

Reply:
Wait, this is in a test class... why is this if guarded at all?

Reply:
Good point. Would you want the call site to be changed too?
```
currently:
#if UNITY_EDITOR || DEVELOPMENT_BUILD
        /// <summary>
        /// Enables testing of the MessageQueueContainer
        /// </summary>
        /// <param name="enabled"></param>
        public void SetTestingState(bool enabled)
        {
            m_IsTestingEnabled = enabled;
        }
#endif
```

Reply:
I think it is because of this build option set within [BuildMultiprocessTestPlayer.BuildPlayer](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/blob/f1a070691e00651a0e356cdb339b7a510cab7278/testproject/Assets/Tests/Runtime/MultiprocessRuntime/Helpers/BuildMultiprocessTestPlayer.cs#L73) when building a "release" version of the clients?

I had to do the [exact same thing in Zain's PR ](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/1114/files)to test the release version of the MultiProcess client build.

Reply:
It looks like (first pass glance) if we migrate the following:
ExecuteStepInContext,TestCoordinator, GameObjectPool, CustomPrefabSpawnerForPerformanceTests, CallbackComponent, and MultiprocessOrchestration
Out of the MultiProcess stuff into a more neutral location then we would not need to include the test assemblies when building the Multiprocess client.

Reply:
MultiprocessTests is only FOR tests - why put in all the work to make it work with builds that aren't test builds? I think this location should drop its guards, and the target function should change to be INCLUDE_UNITY_TESTS, which is also always set in the editor.

Reply:
It is only for tests indeed. The catch is that the MultiProcess Client depends upon classes that are in the MultiProcessTests assembly, and so when you try to build a release version of the MultiProcess Client (i.e. INCLUDE_UNITY_TESTS, UNITY_EDITOR, and DEVELOPMENT_BUILD will not be defined under that one specific scenario and you will get a compilation error when trying to build a RELEASE version of the MultiProcess client.

Reply:
I cannot drop the guards at this location (`Tests/Runtime/RpcQueueTests.cs`) as this result in an immediate build failure on opening the Editor: 
```
/Users/jeffrey.rainy/code/com.unity.multiplayer.mlapi/com.unity.netcode.gameobjects/Tests/Runtime/RpcQueueTests.cs(113,31): error CS1061: 'MessageQueueContainer' does not contain a definition for 'SetTestingState' and no accessible extension method 'SetTestingState' accepting a first argument of type 'MessageQueueContainer' could be found (are you missing a using directive or an assembly reference?)
```

Given that a fix for this is already in Noel's/Zain's PR, and that it affects a bit larger than I expected, I won't be merging this PR. The discussion was quite instructive and, locally, I've got a way to build.